client/my-sites/checkout/checkout-system-decider.js

```diff
@@ -100,6 +101,23 @@ export default function CheckoutSystemDecider( {
 		}
 	}, [ reduxDispatch, checkoutVariant, product ] );
 
+	useEffect( () => {
+		if ( 'composite-checkout' === checkoutVariant && isLoggedOutCart ) {
+			initGoogleRecaptcha(
+				'g-recaptcha',
+				'calypso/checkout/pageLoad',
+				config( 'google_recaptcha_site_key' )
+			).then( ( result ) => {
+				if ( ! result ) {
+					return;
+				}
+
+				const { dispatch } = defaultRegistry;
+				dispatch( 'wpcom' ).setRecaptchaClientId( parseInt( result.clientId ) );
+			} );
+		}
+	}, [ cart, cart.currency ] );
```

Comment:
I think you'll need to add `isLoggedOutCart` and `checkoutVariant` to this dependency array.